custom/icds_reports/static/js/directives/indie-map/indie-map.directive.js

```diff
@@ -69,9 +69,10 @@ function IndieMapController($scope, $compile, $location, $filter, storageService
             }
             if (isMobile) {
                 // scale maps based on space available on device.
-                var headerHeight = $('#map-chart-header').height();
-                // take window height and subtract height of the header plus 44px of additional padding / margins
-                var availableHeight = window.innerHeight - headerHeight - 44;
+                var headerHeight = $('#map-chart-header').height() + $('#page-heading').innerHeight();
+                var legendHeight = 145; // fixed legend height to calculate available space for map
```

Comment:
is this always valid? I'm not sure, but I believe that some legends might have longer text or a different number of colors. have you tested on all reports?

Reply:
I have seen it in all reports. This is the maximum size it can get. I have taken the size of the legend having longest text and 3 colors (Maximum number of colors in all the reports)

Longest text and 3 colors:
![Screenshot from 2020-02-18 16-27-27](https://user-images.githubusercontent.com/56248457/74730190-b29f8b00-526b-11ea-8ec9-cc64d8171f7e.png)

Short text and single color:
![Screenshot from 2020-02-18 16-27-53](https://user-images.githubusercontent.com/56248457/74730236-c3e89780-526b-11ea-90a3-945eca4ad48d.png)

Reply:
perfect, thank you